### DISCUSSION TITLE
Repeat & optional function in buildquery

### DISCUSSION THREAD

**Discussion by anandsubbu007** [isar/isar#248] [Help]
How to use repeat & optional function in buildquery?

**Comment by simc** [comment]
It's not needed there since you can provide dynamic data. There is nothing special about `.optional()` and `.repeat()`. They are both not executed in the database but in Dart.

**Comment by anandsubbu007** [comment]
may I know how to use filter conditions dynamically?
like I need to find items of product where brand contains in the list of the brand if brand list is empty return all.

**Reply by simc** [reply]
Is there a reason you don't use the static extension API? Like `col.filter().brandEqualTo() ...`

**Reply by anandsubbu007** [reply]
how i i use if condition for query series?

like I only need brandEqualTo() if brandFilter is not null

**Reply by simc** [reply]
Using `.optional`:
```dart
col.filter()
  .optional(
    brandFilter != null,
    (q) => q.brandEqualTo('your brand')
  ).findAll()
```
